app/instrumentation-tests/src/org/commcare/androidTests/DateWidgetsTests.kt

```diff
@@ -0,0 +1,208 @@
+package org.commcare.androidTests
+
+import android.util.Log
+import android.widget.DatePicker
+import androidx.test.espresso.Espresso.*
+import androidx.test.espresso.action.ViewActions
+
+import androidx.test.espresso.contrib.PickerActions
+import androidx.test.espresso.matcher.ViewMatchers
+import androidx.test.espresso.matcher.ViewMatchers.*
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.filters.LargeTest
+
+import org.commcare.annotations.BrowserstackTests
+import org.commcare.dalvik.R
+import org.commcare.utils.InstrumentationUtility
+import org.commcare.utils.isPresent
+import org.hamcrest.Matchers
+import org.hamcrest.Matchers.allOf
+
+import org.junit.Assert.*
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.text.SimpleDateFormat
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
+import java.util.*
+import kotlin.math.absoluteValue
+
+
+
+@RunWith(AndroidJUnit4::class)
+@LargeTest
+@BrowserstackTests
+class DateWidgetsTests: BaseTest() {
+    companion object {
+        const val CCZ_NAME = "date_widgets_tests.ccz"
+        const val APP_NAME = "Date Widgets"
+        val listOfNepaliMonths = listOf("Baishakh","Jestha","Ashadh","Shrawan",
+                            "Bhadra","Ashwin","Kartik","Mangsir","Poush","Magh",
+                            "Falgun","Chaitra")
+        val listOfEthiopianMonths = listOf("Säne","Hämle","Nähäse","P’agume",
+            "Mäskäräm","T’ïk’ïmt","Hïdar","Tahsas","T’ïr","Yäkatit",
+            "Mägabit","Miyaziya")
+
+    }
+
+    @Before
+    fun setup() {
+        installApp(APP_NAME, CCZ_NAME)
+        InstrumentationUtility.login("test1", "123")
+    }
+
+    /**
+     * Adding the teardown method so the tests doesnot fail after only thefirst execution, when executed as a whole.
+     */
+
+    @Test
+    fun testNepalDateWidgets(){
+
+        InstrumentationUtility.openModule(0)
+        onView(withSubstring("This form will test the Nepal date widget.")).isPresent()
+        InstrumentationUtility.nextPage()
+        InstrumentationUtility.nextPage()
+
+        for (i in 1..12){
+            val month_text = InstrumentationUtility.getText(onView(withId(R.id.monthtxt)))
+            if (month_text != null) {
+                assertTrue(listOfNepaliMonths.contains(month_text))
```

Comment:
this doesn't ensure that the months appear in right order. I think a better approach would be - 
1. Set date to say 11-01-2022 for example at start
2. Loop as `for (i in 0..11)`
3. Check whether `listOfNepaliMonths[i]` is present on the screen

This way I don't think you need to use the `InstrumentationUtility.getText` method as well as you can just check for `listOfNepaliMonths[i]` to be present on screen

Reply:
implemented the logic that reads the current month, gets the current index
in the list, rotates the list starting from current month, loops through the list and asserts if months are present in order. I also printed them to verify if the list rotation was working correctly, before implementing.